draft-ietf-quic-transport.md

```diff
@@ -4106,6 +4102,12 @@ A client MUST NOT include an original connection ID, a stateless reset token, or
 a preferred address.  A server MUST treat receipt of any of these transport
 parameters as a connection error of type TRANSPORT_PARAMETER_ERROR.
 
+max_connection_ids (0x000e):
+
+: The maximum number of connection IDs that the peer is willing to store.
+  This value includes only connection IDs sent in NEW_CONNECTION_ID frames.
+  If this parameter is absent, a default of 0 is assumed.
```

Comment:
I'd really like to be able to not send this.  Is 0 really the right answer?

Reply:
We had a recommended value previously, let's pull that value over and have the default value conversation separately?

Reply:
We discussed this before in https://github.com/quicwg/base-drafts/pull/1998#discussion_r233121475. We're using 0 as the default value for all flow control and stream limit values, so at least this seems consistent.

In Tokyo we had disagreement about what a reasonable default should be. @ianswett advocated for 2 or 3, if I remember correctly, while other people want 8. One reason for moving forward with this PR was exactly the fact that people have different use cases in mind, and there's no one-size-fits-all value.

Reply:
What if an endpoint set both "migration disabled = 0" (default) and "max CID=0" (default)? Should that trigger a protocol error?

Reply:
By the way, we already use no zero defaults for some parameters: ack_delay_exponent defaults to 3, and max_ack_delay defaults to 25. There is no rule that says that forbids us to set a non zero default for the proposed MAX CID parameter. I would think that 3 is a good number.

Reply:
In HTTP, we use zero defaults for everything because those values apply until the SETTINGS frame arrives.  In QUIC, however, you can't send anything until transport parameters are in hand.  Any value can be a reasonable default.  (Note that the requirement not to reduce after accepting 0-RTT will apply here, though.)

Reply:
This is one where reducing after 0-RTT is perfectly OK.  It's not a hard limit, it's more like a guideline.  We should allow an exception here.

Reply:
True -- we have an exemption for the one HTTP setting which is purely advisory.  Since this is a suggestion, reducing it wouldn't necessarily hurt anything.  (But there's no prohibition on the sender to not blow up on exceeding the limit, either.)

Reply:
I am a little hazy on the proper behavior for NAT rebinding vs. migration, but how is communicating a zero here different from disable_migration? Do we still need that TP?

Reply:
Yes, because you might want to rotate CIDs on a connection that isn't actively migrating.  Whether you'll survive NAT rebinding depends on the server infrastructure -- a 2-tuple load balancer will survive a port change, but a 4-tuple load balancer can't guarantee it.  An IP+port change isn't guaranteed to survive on anything but a CID-aware load balancer.